### PR TITLE
Add now_utc

### PR DESCRIPTION
**Pull request by alissapajer**
Requires https://github.com/precog/mimir/pull/1070

### CODE REVIEW

File: qscript/src/main/scala/quasar/qscript/MapFuncCore.scala [hunk 1]
```diff
@@ -869,11 +873,14 @@ object MapFuncsCore {
   @Lenses final case class ToTimestamp[T[_[_]], A](a1: A) extends Unary[T, A]
   @Lenses final case class ToLocal[T[_[_]], A](a1: A, a2: A) extends Binary[T, A]
   @Lenses final case class ToOffset[T[_[_]], A](a1: A, a2: A) extends Binary[T, A]
-  /** Fetches the [[quasar.Type.OffsetDateTime]] for the current instant in time, in UTC. */
+
+  /** Fetches the [[quasar.Type.LocalDateTime]] for the current instant in time, in the local timezone. */
```
Comment (rintcius) [inline qscript/src/main/scala/quasar/qscript/MapFuncCore.scala:877]: Should we add a TODO here that this should be an `OffsetDateTime`? Looks to me that's the proper type to use here, a `LocalDateTime` doesn't seem to be that useful as a type for `now`. Also see eg https://www.postgresql.org/docs/8.2/functions-datetime.html : `now()` - `timestamp with time zone`. 

Having it return this type would also make it in line with `NowTime` and `NowDate`.

Comment (wemrysi) [reply qscript/src/main/scala/quasar/qscript/MapFuncCore.scala:877]: I think having all of these include timezones eventually sounds good. Also, `mimir` currently produces `Local*` variants for `NowTime` and `NowDate`, despite the docs, so even if they aren't that useful they're at least consistent 😄

**Review by wemrysi — APPROVED** [review]

**Comment by wemrysi** [comment]
@alissapajer I think this PR is `breaking` since it introduces a new `MapFuncCore` constructor.

File: qscript/src/main/scala/quasar/qscript/MapFuncCore.scala (hunk 1 above)
Comment (wemrysi) [reply qscript/src/main/scala/quasar/qscript/MapFuncCore.scala:877]: @rintcius I added a clubhouse ticket regarding the "now" variants including offsets.

Comment (rintcius) [reply qscript/src/main/scala/quasar/qscript/MapFuncCore.scala:877]: Ah right, good to fix the docs and add a TODO referring to that ticket https://app.clubhouse.io/data/story/12764/now-sql-2-functions-should-include-timezone-offset to decrease chances that someone working on something related in the future misses this.

**Pull request merged**